File: python/ch01/14.declarative.py
from langchain_openai.chat_models import ChatOpenAI
from langchain_core.prompts import ChatPromptTemplate

# 구성 요소

template = ChatPromptTemplate.from_messages(
    [
        ('system', '당신은 친절한 어시스턴트입니다.'),
        ('human', '{question}'),
    ]
)

model = ChatOpenAI()

# 연산자 | 로 결합한다

chatbot = template | model

# 사용한다

response = chatbot.invoke({'question': '거대 언어 모델은 어디서 제공하나요?'})
print(response.content)

File: javascript/ch01/14.declarative.js
import { ChatOpenAI } from '@langchain/openai';
import { ChatPromptTemplate } from '@langchain/core/prompts';

// 구성 요소

const template = ChatPromptTemplate.fromMessages([
  ['system', '당신은 친절한 어시스턴트입니다.'],
  ['human', '{question}'],
]);

const model = new ChatOpenAI({
  model: 'gpt-3.5-turbo',
  streaming: true, // 스트리밍 활성화
});

// 함수에서 구성 요소 결합

const chatbot = template.pipe(model);

// 사용한다

const response = await chatbot.invoke({
  question: '거대 언어 모델은 어디서 제공하나요?',
});

console.log(response);